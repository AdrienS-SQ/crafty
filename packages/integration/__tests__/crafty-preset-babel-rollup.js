/* global describe, it, expect */
const path = require("path");

const { run, cleanDist, exists, readFile } = require("../utils");

jest.setTimeout(10000);

it("Compiles JavaScript with rollup", async () => {
  const dir = path.join(
    __dirname,
    "../fixtures/crafty-preset-babel-rollup/compiles"
  );
  await cleanDist(dir);

  const result = await run(dir, ["run", "default"]);

  expect(result).toMatchSnapshot();

  expect(exists(dir, "dist/js/myBundle.min.js")).toBeTruthy();
  expect(exists(dir, "dist/js/myBundle.min.js.map")).toBeTruthy();

  expect(readFile(dir, "dist/js/myBundle.min.js")).toMatchSnapshot();
});

it("Fails gracefully on broken markup", async () => {
  const dir = path.join(
    __dirname,
    "../fixtures/crafty-preset-babel-rollup/fails"
  );
  await cleanDist(dir);

  const result = await run(dir, ["run", "default"]);

  expect(result).toMatchSnapshot();

  expect(exists(dir, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(exists(dir, "dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints JavaScript with rollup", async () => {
  const dir = path.join(
    __dirname,
    "../fixtures/crafty-preset-babel-rollup/lints"
  );
  await cleanDist(dir);

  const result = await run(dir, ["run", "default"]);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(exists(dir, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(exists(dir, "dist/js/myBundle.min.js.map")).toBeFalsy();
});
