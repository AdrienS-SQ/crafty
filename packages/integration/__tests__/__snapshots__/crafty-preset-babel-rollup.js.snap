// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiles JavaScript with rollup 1`] = `
Object {
  "status": 0,
  "stderr": "",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles JavaScript with rollup 2`] = `
"function _classCallCheck(e,t){if(!(e instanceof t))throw new TypeError(\\"Cannot call a class as a function\\")}function _defineProperties(e,t){for(var n=0;n<t.length;n++){var o=t[n];o.enumerable=o.enumerable||!1,o.configurable=!0,\\"value\\"in o&&(o.writable=!0),Object.defineProperty(e,o.key,o)}}function _createClass(e,t,n){return t&&_defineProperties(e.prototype,t),n&&_defineProperties(e,n),e}function _inherits(e,t){if(\\"function\\"!=typeof t&&null!==t)throw new TypeError(\\"Super expression must either be null or a function\\");e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,writable:!0,configurable:!0}}),t&&_setPrototypeOf(e,t)}function _getPrototypeOf(e){return(_getPrototypeOf=Object.setPrototypeOf?Object.getPrototypeOf:function(e){return e.__proto__||Object.getPrototypeOf(e)})(e)}function _setPrototypeOf(e,t){return(_setPrototypeOf=Object.setPrototypeOf||function(e,t){return e.__proto__=t,e})(e,t)}function _assertThisInitialized(e){if(void 0===e)throw new ReferenceError(\\"this hasn't been initialised - super() hasn't been called\\");return e}function _possibleConstructorReturn(e,t){return!t||\\"object\\"!=typeof t&&\\"function\\"!=typeof t?_assertThisInitialized(e):t}var MyComponent=function(e){function t(){var e,n;_classCallCheck(this,t);for(var o=arguments.length,r=new Array(o),i=0;i<o;i++)r[i]=arguments[i];return(n=_possibleConstructorReturn(this,(e=_getPrototypeOf(t)).call.apply(e,[this].concat(r)))).handleClick=function(){n.setState({hey:!0})},n}return _inherits(t,React.Component),_createClass(t,[{key:\\"render\\",value:function(){return React.createElement(\\"div\\",{onClick:this.handleClick},\\"Hey \\",this.state.hey)}}]),t}(),newStuff=function(){function e(){_classCallCheck(this,e),console.log(\\"init class\\")}return _createClass(e,[{key:\\"method\\",value:function(){console.log(MyComponent)}}]),e}();export default newStuff;
//# sourceMappingURL=myBundle.min.js.map
"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 1,
  "stderr": "[js_myBundle] (!) Errors were found
js/script.js
[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...

__PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/fails/js/script.js
  15:1  error  Parsing error: Unexpected token

  13 | 
  14 | }
> 15 | 
     | ^

✖ 1 problem (1 error, 0 warnings)

",
}
`;

exports[`Lints JavaScript with rollup 1`] = `
Object {
  "status": 1,
  "stderr": "[js_myBundle] (!) Errors were found
js/script.js
[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
  "stdout": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/lints/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...

__PATH__/packages/integration/fixtures/crafty-preset-babel-rollup/lints/js/script.js
   3:24  error    Multiple spaces found before '\\"./Component.js\\"'                                no-multi-spaces
   5:10  error    'test' is defined but never used                                               no-unused-vars
   5:19  error    A space is required after ','                                                  comma-spacing
   6:1   error    Expected indentation of 2 spaces but found 3                                   indent
  11:5   warning  Unexpected console statement                                                   no-console
  15:5   warning  Unexpected console statement                                                   no-console
  15:19  error    Multiple spaces found before '\\"some method\\"'                                   no-multi-spaces
  15:33  error    There should be no space before ','                                            comma-spacing
  19:3   error    Opening curly brace does not appear on the same line as controlling statement  brace-style
  20:14  error    Expected '===' and instead saw '=='                                            eqeqeq

✖ 10 problems (8 errors, 2 warnings)
  6 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;
